conans/client/downloaders/download_cache.py

```diff
@@ -120,6 +120,9 @@ def update_backup_sources_json(cached_path, conanfile, urls):
 
         try:
             summary_key = str(conanfile.ref)
+            if set(summary["references"].keys()).difference({"unknown", summary_key}):
```

Comment:
Possible false positives:
- Different packages with different user/channel, but downloading the same source
- Packages downloading sources of a vendored dependency that is also a Conan package elsewhere
- Package re-names (like  changing style like ``my_pkg`` to ``mypkg`` or ``mypkg++`` to ``mypkgpp`` to comply with Conan recommendations.